web/modules/custom/dbcdk_community/src/Plugin/Block/FlaggedContentDetails.php

```diff
@@ -0,0 +1,189 @@
+   *   The repository to use when retrieving flaggable content.
+   * @param ProfileApi $profile_api
+   *   The profile api to use when retrieving information about content owners.
+   * @param UrlGeneratorInterface $url_generator
+   *   The url generator to use when generating links to the biblo.dk site.
+   */
+  public function __construct(
+        array $configuration,
+        $plugin_id,
+        $plugin_definition,
+        LoggerInterface $logger,
+        FlaggableContentRepository $repository,
+        ProfileApi $profile_api,
+        UrlGeneratorInterface $url_generator
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->logger = $logger;
+    $this->flaggableContentRepository = $repository;
+    $this->profileApi = $profile_api;
+    $this->urlGenerator = $url_generator;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    /* @var \Drupal\Core\Logger\LoggerChannelFactory $logger_factory */
+    $logger_factory = $container->get('logger.factory');
+
+    // Setup url generation for flaggable content.
+    /* @var \Drupal\Core\Config\Config $config */
+    $config = $container->get('config.factory')->get('dbcdk_community.settings');
+    $base_url = $config->get('community_site_url');
+    $url_generator = new ModelUrlGenerator();
+    $url_generator->registerClassGenerator(
+      'DBCDK\CommunityServices\Model\Post',
+      new PropertyUrlGenerator(
+        $base_url . $config->get('community_site_post_url_pattern')
+      )
+    );
+    $url_generator->registerClassGenerator(
+      'DBCDK\CommunityServices\Model\Comment',
+      new PropertyUrlGenerator(
+        $base_url . $config->get('community_site_comment_url_pattern')
+      )
+    );
+
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $logger_factory->get('DBCDK Community Service'),
+      $container->get('dbcdk_community.content.flaggable_content_repository'),
+      $container->get('dbcdk_community.api.profile'),
+      $url_generator
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $flag_id = $this->getContextValue('flag_id');
+
+    try {
+      $content = $this->flaggableContentRepository->getContentById($flag_id);
+      $profile = $this->profileApi->profileFindById($content->getCreatorId());
+
+      $profile_link = Link::createFromRoute(
+        $profile->getUsername(),
+        'page_manager.page_view_dbcdk_community_profile',
+        ['username' => $profile->getUsername()]
+      );
+
+      $url = $this->urlGenerator->generate($content->getObject());
+      $content_link = Link::fromTextAndUrl($url, Url::fromUri($url));
+
+      $rows = [
+        [$this->t('Content'), $content->getContent()],
+        [$this->t('Created on'), $content->getTimeCreated()->format('H:i d-m-Y')],
```

Comment:
I agree. Should we create a common format for dates + times just like #38 defines one for dates only?